vizro-ai/tests/unit/vizro-ai/dashboard/_response_models/test_controls.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+import vizro.models as vm
+from vizro.managers import model_manager
+from vizro.models import VizroBaseModel
+from vizro_ai.dashboard._response_models.controls import ControlPlan, _create_filter, _create_filter_proxy
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+# Needed for testing control creation.
+model_manager.__setitem__("bar_chart", VizroBaseModel)
+
+
+class TestFilterProxyCreate:
+    """Tests filter proxy creation."""
+
+    def test_create_filter_proxy_validate_targets(self, df_cols, df_schema, controllable_components):
+        filter_proxy = _create_filter_proxy(df_cols, df_schema, controllable_components)
+        with pytest.raises(ValidationError, match="targets must be one of"):
+            filter_proxy(targets=["population_chart"], column="a")
+
+    def test_create_filter_proxy_validate_targets_not_empty(self, df_cols, df_schema, controllable_components):
+        filter_proxy = _create_filter_proxy(df_cols=df_cols, df_schema=df_schema, controllable_components=[])
+        with pytest.raises(ValidationError):
+            filter_proxy(targets=[], column="a")
+
+    def test_create_filter_proxy_validate_columns(self, df_cols, df_schema, controllable_components):
+        filter_proxy = _create_filter_proxy(df_cols, df_schema, controllable_components)
+        with pytest.raises(ValidationError, match="column must be one of"):
+            filter_proxy(targets=["bar_chart"], column="x")
+
+    def test_create_filter_proxy(self, df_cols, df_schema, controllable_components, expected_filter):
+        filter_proxy = _create_filter_proxy(df_cols, df_schema, controllable_components)
+        result = filter_proxy(targets=["bar_chart"], column="a")
+
+        assert result.dict(exclude={"id": True}) == expected_filter.dict(exclude={"id": True})
+
+
+class TestControlPlan:
+    """Test control plan."""
+
+    def test_control_plan_invalid_df_name(self, fake_llm_filter, df_metadata):
+        control_plan = ControlPlan(
+            control_type="Filter",
+            control_description="Create a filter that filters the data based on the column 'a'.",
+            df_name="population_chart",
+        )
+        default_control = control_plan.create(
```

Comment:
For future reference, or the next PR: I think it would be awesome if we could test everything around the creation (proxy and final object) at this interface level. That makes it easier to read, and allows to try out all the combinations that work and do not work. Not for now, but for the future, wdyt?

Reply:
Ideas would be: parametrize along the three inputs: model, controllable components and metadata, and think about the cases that could happen: e.g. i have these conrollable components, and my models returns this, i have the same controllable components, but my model returns something else. Etc etc.

Does that make sense?

Reply:
i've tested both valid and invalid scenarios with `.create()`. We can expand and add more tests cases in the next PR. I'll create a ticket for it.